Adyen/Analytics/AnalyticsProvider/AnalyticsProvider.swift

```diff
@@ -66,28 +81,67 @@ internal final class AnalyticsProvider: AnalyticsProviderProtocol {
     }
 
     // MARK: - Internal
-    
-    internal func fetchAndCacheCheckoutAttemptIdIfNeeded() {
-        fetchCheckoutAttemptId { _ in /* Do nothing, the point is to trigger the fetching and cache the value  */ }
-    }
 
-    internal func fetchCheckoutAttemptId(completion: @escaping (String?) -> Void) {
-        guard configuration.isEnabled else {
+    internal func fetchCheckoutAttemptId(with flavor: TelemetryFlavor, additionalFields: AdditionalAnalyticsFields?) {
+        guard configuration.isEnabled, configuration.isTelemetryEnabled else {
             checkoutAttemptId = "do-not-track"
-            completion(checkoutAttemptId)
             return
         }
+        if case .dropInComponent = flavor { return }
+        
+        let telemetryData = TelemetryData(flavor: flavor,
+                                          additionalFields: additionalFields,
+                                          context: configuration.context)
 
-        let checkoutAttemptIdRequest = CheckoutAttemptIdRequest()
+        let checkoutAttemptIdRequest = CheckoutAttemptIdRequest(data: telemetryData)
 
         uniqueAssetAPIClient.perform(checkoutAttemptIdRequest) { [weak self] result in
             switch result {
             case let .success(response):
                 self?.checkoutAttemptId = response.identifier
-                completion(response.identifier)
             case .failure:
-                completion(nil)
+                self?.checkoutAttemptId = nil
             }
         }
     }
+    
+    internal func send(event: AdyenAnalytics.Event) {
+        events.append(event)
+    }
+    
+    internal func send(log: AdyenAnalytics.Log) {
+        logs.append(log)
+    }
+    
+    internal func send(error: AdyenAnalytics.Error) {
+        errors.append(error)
+        sendAll()
+    }
+    
+    private func setupTimer() {
+        let timer = Timer(timeInterval: 10, target: self, selector: #selector(sendAll), userInfo: nil, repeats: true)
+        timer.tolerance = 1
+        RunLoop.current.add(timer, forMode: .common)
+    }
+    
+    @objc private func sendAll() {
+        guard configuration.isEnabled,
+              let checkoutAttemptId else { return }
+        var request = AdyenAnalyticsRequest(checkoutAttemptId: checkoutAttemptId)
+        
+        request.events = events
+        request.logs = logs
+        request.errors = errors
+        
+        apiClient.perform(request) { [weak self] _ in
+            guard let self else { return }
+            self.clearAll()
```

Comment:
This could lead to losing some events that arrived while the request was going on.

Maybe temporarily hold onto the events (`var eventsBeforeApiCall`) and clear the current ones before making the api call (`clearAll`) and if it fails append the saved ones to the list.

Reply:
Will work on this area on next PR, would have been better not to include this code here but yeah.